ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/daml_kvutils.proto

```diff
@@ -184,20 +190,35 @@ message DamlTransactionBlindingInfo {
 // A transaction rejection entry.
 message DamlTransactionRejectionEntry {
   DamlSubmitterInfo submitter_info = 1;
-
   reserved 5; // was reason.maximum_record_time_exceeded
 
   oneof reason {
+    // errors used in v1 which are still used in v2
+    InvalidLedgerTime invalid_ledger_time = 9;
+    SubmitterCannotActViaParticipant submitter_cannot_act_via_participant = 8;
+
+    // v1 reasons (these are DEPRECATED)
     Inconsistent inconsistent = 2;
     Disputed disputed = 3;
     ResourcesExhausted resources_exhausted = 4;
     Duplicate duplicate_command = 6;
     PartyNotKnownOnLedger party_not_known_on_ledger = 7;
-    SubmitterCannotActViaParticipant submitter_cannot_act_via_participant = 8;
-    InvalidLedgerTime invalid_ledger_time = 9;
+
+    // v2 reasons
+    ValidationFailure validation_failure = 10;
+    DuplicateKeys internal_duplicate_keys = 11;
```

Comment:
Was there a reason to diverge from the respective Scala rejection names such as `InternallyInconsistentTransaction.DuplicateKeys`? In order to be consistent with the Scala class names I'd suggest using `internally_duplicate_keys`, `internally_inconsistent_keys`, etc. as field names here, or even better, create a separate message mirroring the Scala class and declare two separate fields for rejections because of internally vs. externally inconsistent transactions so that we keep this message easier to read.
E.g., this message could look something like:
```protobuf
message DamlTransactionRejectionEntry {
  ...
  oneof reason {
    <common rejection reasons>
    //
    // Rejections used by participant.state.v1 API.
    // Note that these are deprecated.
    //
    Inconsistent inconsistent = 2;
    Disupted disupted = 3;
    ...

    //
    // Rejections used by participant.state.v2 API.
    //
    ValidationFailure validation_failure = 10;
    InconsistentTransaction internally_inconsistent_transaction = ...;
    InconsistentTransaction externally_inconsistent_transaction = ...;
    ...
  }
}
```
Another option to consider is to introduce a separate message encompassing all rejection reasons specific to `participant.state.v2`.
@nicu-da What do you think?

Reply:
Makes sense, I like a lot more the header field for versioning. Will update that and naming.

Reply:
I think mirroring the scala namespacing for "external" vs "internal" doesn't really make sense, because internally we have only "inconsistent" and "duplicate" keys while externally we also have "inconsistent contracts" so the models don't really map 1:1.